lib/controllers/root-controller.js

```diff
@@ -488,8 +497,30 @@ export default class RootController extends React.Component {
 
   async viewChangesForCurrentFile(stagingStatus) {
     const editor = this.props.workspace.getActiveTextEditor();
-    const absFilePath = editor.getPath();
+    const absFilePath = await realPath(editor.getPath());
     const repoPath = this.props.repository.getWorkingDirectoryPath();
+    if (repoPath === null) {
+      const [projectPath] = this.props.project.relativizePath(editor.getPath());
+      const notification = this.props.notificationManager.addInfo(
+        "Hmm, we don't have anything to compare this to.",
+        {
+          description: 'Create a Git repository in order to view changes to the files in your project.',
+          dismissable: true,
+          buttons: [{
+            className: 'btn btn-primary',
+            text: 'Create a repository now',
+            onDidClick: async () => {
+              notification.dismiss();
+              const created = await this.initializeRepo(projectPath);
+              // If the user confirmed repository creation,
+              // retry the operation that got them here in the first place
+              if (created) { this.viewChangesForCurrentFile(stagingStatus); }
```

Comment:
I think it's possible for the user to delete the suggested project path and initialize a different path instead, in which case the retry would... show this prompt again.

Maybe `initializeRepo` could resolve with the initialized path instead, and the retry could only be done if it's the same as projectPath... ?